### PR TITLE
 [EI-1692]Add Unit test for X-forwarded header fix

### PR DESCRIPTION
**Pull request by vinok88**
Issue - wso2/product-ei#1692

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: modules/extensions/src/test/java/org/apache/synapse/mediators/throttle/ThrottleMediatorTest.java [hunk 1]
```diff
@@ -347,6 +352,41 @@ public void testConcurrencyThrottlingMediationWithInLineXML() throws Exception {
...
+                throttleMediator.mediate(synCtx);
+                Thread.sleep(1000);
+            } catch (Exception e) {
+
+                if (i == 3) {
+                    assertTrue(e.getMessage().lastIndexOf("IP_BASE") > 0);
+                } else if (i == 4) {
+                    assertTrue(e.getMessage().lastIndexOf("IP_BASE") > 0);
+                } else if (i == 5) {
+                    assertTrue(e.getMessage().lastIndexOf("IP_BASE") > 0);
```
Comment (Nirothipan) [inline modules/extensions/src/test/java/org/apache/synapse/mediators/throttle/ThrottleMediatorTest.java]: Isn't it better to add some assert messages here?

**Comment by prabushi** [comment]
Shall we adhere to the commit message convention?

File: modules/extensions/src/test/java/org/apache/synapse/mediators/throttle/ThrottleMediatorTest.java [hunk 2]
```diff
@@ -347,6 +352,37 @@ public void testConcurrencyThrottlingMediationWithInLineXML() throws Exception {
 
     }
 
+    /**
+     * Test the XForwarded header based throttling.
+     * @throws Exception
+     */
+    public void testMediateWithXForwardedHeader() throws Exception {
+        ByteArrayInputStream in = new ByteArrayInputStream(NEW_POLICY.getBytes());
```
Comment (wggihan) [inline modules/extensions/src/test/java/org/apache/synapse/mediators/throttle/ThrottleMediatorTest.java]: I couldn't find any code block that closes this input stream.

File: modules/extensions/src/test/java/org/apache/synapse/mediators/throttle/ThrottleMediatorTest.java [hunk 3]
```diff
@@ -347,6 +352,37 @@ public void testConcurrencyThrottlingMediationWithInLineXML() throws Exception {
...
+        });
+        headers.put(NhttpConstants.HEADER_X_FORWARDED_FOR, "192.168..215");
+        SynapseConfiguration synCfg = new SynapseConfiguration();
+        synCtx.setConfiguration(synCfg);
+        for (int i = 0; i < 6; i++) {
+            try {
+                throttleMediator.mediate(synCtx);
+                Thread.sleep(1000);
+            } catch (Exception e) {
+
```
Comment (wggihan) [inline modules/extensions/src/test/java/org/apache/synapse/mediators/throttle/ThrottleMediatorTest.java]: Remove these empty lines.

**Pull request merged**